multipatch_analysis/ui/acq4_modules.py

```diff
@@ -56,3 +71,266 @@ def load_from_dm_clicked(self):
+            p = cell[1]
+            for i in items:
+                try:
+                    if i.opts['handle'].name() == self.image_20:
+                        image = i
+                except AttributeError:
+                    pass
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).x())
+            print(markers[0].graphicsItem().mapToItem(image.graphicsItem(),pg.Point(*p[:2])).y())
+
+        
+
+    
+
+    def load_clicked(self):
+        """
+        Checks that directory is pointed at a slice
+        Check LIMS for 20x biocytin image and if 20x image is saved locally.
+        If image not saved locally then copies file from LIMS.
+        Opens button to create a JSON of the cell locations
+        """
+        self.base_dir = self.mosaic_editor.ui.fileLoader.baseDir()
+        if self.base_dir.info()['dirType'] == 'Slice':
+            self.base_path = self.base_dir.name()
+            self.slice_name = self.base_dir.info()['specimen_ID']
+            self.slice_id = lims.specimen_id_from_name(self.slice_name)
+            clusters = lims.cell_cluster_ids(self.slice_id)
+            """if len(clusters) == 0:
+                                                    raise Exception("No Clusters Recorded in LIMS")"""
+            try:
+                aff_image_path = lims.specimen_20x_image(self.slice_name)
+                #check the image file path
+                safe_aff_image_path = lims.lims.safe_system_path(aff_image_path)
+
+                if os.path.exists(safe_aff_image_path) == True:
+                    print('File Exists in LIMS')
+                    aff_image_name = aff_image_path.split("/")[-1]
+                    jpg_image_name = aff_image_name.split(".aff")[0] + ".jpg"
+
+                    image_service_url = "http://lims2/cgi-bin/imageservice?path="
+                    zoom = 5        #need to check that this ACQ4 can support this size
+
+                    full_url = image_service_url + aff_image_path + "&zoom={}".format(str(zoom))
+                    print(full_url)
+                    save_path = self.base_path + '/' + jpg_image_name
+                    safe_save_path = lims.lims.safe_system_path(save_path)
+
+                    if os.path.exists(safe_save_path) == True:
+                        print("20x already moved")
+
+                    elif os.path.exists(safe_save_path) == False:
+                        image = urllib.URLopener()
+                        image.retrieve(full_url, safe_save_path)
+                        print("20x image moved")
+                    self.image_20 = safe_save_path
+                        
+                else:
+                    raise Exception("Couldn't find image in LIMS. Check you have the selected the correct slice.")
+                
+            except KeyError:
+                raise Exception('No Slice Selected')
+        else:
+            raise Exception('No Slice Selected')
+       
```

Comment:
After you get to this point, ask the mosaiceditor to load the file?

Reply:
I don't think I want to have it load the file. Because when you load the site mosaic it dumps everything on the canvas

Reply:
after the file gets moved from lims to data folder it doesn't show up in the file loader window. Right clicking, refresh gives an error that "FileHandle" object has no attribute 'ls'

Reply:
Wouldn't the user load the site mosaic first, and then click the 20x button?

Reply:
Maybe? I just changed the button name to 'find 20x' before you made that comment. I was assuming they would look for the 20x before loading the site mosaic because why would you load the mosaic if you didn't have a 20x to tag?

Reply:
..but how do you even know which 20x to load before the mosaic has been loaded?

Reply:
the workflow in my head was the user would choose their slice and assign the base directory to that slice. They would then load the 20x for that slice and then work through each patch site. So then they are only loading the 20x once per slice if there are multiple sites on the slice

Reply:
Yes, you are correct. I'm just confusing myself.